design-proposals/primary_network_binding_for_seamless_migration.md

```diff
@@ -0,0 +1,520 @@
+
+```
+kubectl patch kubevirts -n kubevirt kubevirt --type=json -p=\
+'[{"op": "add", "path": "/spec/configuration/network",   "value": {
+    "binding": {
+        "macvtap": {
+            "domainAttachmentType": "managedMacvtap"
+        }
+    }
+}}]'
+```
+
+#### Other L2 Bindings
+
+The bridge and macvtap bindings have one behavior in common: They
+forward traffic based on L2 (ethernet frames) information only.
+
+Similar bindings may be invented to forward traffic using other technologies.
+Some may be more efficient than others with different complexities but
+with a very similar pros and cons as seen on the previous bridge &
+macvtap bindings, e.g.:
+
+- Passing the IP & route configuration from the pod interface to the VM
+  guest interface is left to the network provider (e.g. DHCPv4, DHCPv6, RA).
+
+Some of the suggested L2 alternatives are:
+
+- OVS bridge
+- eBPF
+- Tap device directly connected to the node, passed into the pod for
+  consumption (saving the veth pair that links the pod interface to the
+  node).
+
+#### Summary
+
+| Feature\\Binding            | Passt                                                                                                           | Bridge[*]                            | Macvtap                                          |
+|:----------------------------|:----------------------------------------------------------------------------------------------------------------|:-------------------------------------|:-------------------------------------------------|
+| Deployment                  | Binding Plugin                                                                                                  | Core                                 | Binding Plugin                                   |
+| IP & Route delivery         | Self (IPv4 & IPV6)                                                                                              | Self for IPv4, Net Provider for IPv6 | Net Provider                                     |
+| Pod Probes Support          | Yes                                                                                                             | No                                   | No                                               |
+| Seamless Live-Migration     | Requires Development                                                                                            | Yes                                  | Yes                                              |
+| Supported Traffic           | TCP/UDP/ICMP (All IP in case seamless migration is resolved)                                                    | IP                                   | IP                                               |
+| Migration Support Indicator | Yes                                                                                                             | Requires Development                 | Yes                                              |
```

Comment:
maybe bridge doesn't require this at all. 

One of the options you present in your enhancement is literally:
> Drop the existing limitation that blocks the migration when bridge
    binding is used with the primary pod network. I.e. always allow
    migration.